codegen-library.sh

```diff
@@ -29,7 +29,7 @@ export GOPATH=$(go_mod_gopath_hack)
 export GOBIN=${GOPATH}/bin # Set GOBIN explicitly as deepcopy-gen is installed by go install.
 export MODULE_NAME=$(go_mod_module_name)
 export CODEGEN_PKG=${CODEGEN_PKG:-$(cd ${REPO_ROOT_DIR}; ls -d -1 ./vendor/k8s.io/code-generator 2>/dev/null || echo ../code-generator)}
-export KNATIVE_CODEGEN_PKG=${KNATIVE_CODEGEN_PKG:-$(cd ${REPO_ROOT_DIR}; ls -d -1 ./vendor/knative.dev/pkg 2>/dev/null || echo ../pkg)}
+export KNATIVE_CODEGEN_PKG=${KNATIVE_CODEGEN_PKG:-$(cd ${REPO_ROOT_DIR}; ls -d -1 ./vendor/knative.dev/pkg 2>/dev/null || echo "${REPO_ROOT_DIR}")}
```

Comment:
I think this will work for the `pkg` repo but seems like it will fail for all other repos. TBH, i don't understand when it is the case where vendor does not have pkg... likely some out of tree integration? pre-gomod?

Given that all other repos in Knative should vendor, I think this is a safe change, and if it breaks someone out of tree they can help make a solution that works for them and us. `../pkg` is not a safe assumption in a go mod world.

/lgtm
/approve